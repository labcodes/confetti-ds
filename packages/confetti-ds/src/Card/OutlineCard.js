import React from "react";
import PropTypes from "prop-types";

import { CardContext } from "./contexts";
import { outlineCardChoices } from "./propTypes";

export default class OutlineCard extends React.Component {
  static propTypes = {
    /** Components that will be rendered in the Outline Card (CardImage, CardHeader, CardAction, etc.) */
    children: PropTypes.node.isRequired,
    /** Sets Outline Card's colors (mostly border colors) based on the chosen palette.  */
    color: PropTypes.oneOf(outlineCardChoices.color).isRequired,
    /** Sets the skin of the rendered Outline Card. */
    skin: PropTypes.oneOf(outlineCardChoices.skin).isRequired,
    /** Sets horizontal layout. It changes the position of CardImage. */
    isHorizontal: PropTypes.bool,
    /** Reduces paddings and margins on Card layout. */
    isCompact: PropTypes.bool,
  };

  static defaultProps = {
    isHorizontal: false,
    isCompact: false,
  };

  render() {
    const { children, color, skin, isCompact, isHorizontal } = this.props;

    return (
      <CardContext.Provider value={{ color, skin, cardType: "outline" }}>
        <article
          className={`lab-card lab-card--outline lab-card--${color} lab-card--${skin}
          ${isCompact ? " lab-card--compact" : ""}
          ${isHorizontal ? " lab-card--horizontal" : ""}
        `}
        >
          {children}
        </article>
      </CardContext.Provider>
    );
  }
}
