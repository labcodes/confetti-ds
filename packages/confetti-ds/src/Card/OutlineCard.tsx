import React from "react";

import { CardContext } from "./contexts";

interface OutlineCardProps {
  /** Components that will be rendered in the Card (CardImage, CardHeader, CardAction, etc.) */
  children: React.ReactNode;
  /** Sets Card's colors (mostly background colors) based on the chosen palette.  */
  color: "mineral" | "teal" | "purple";
  /** Sets the skin of the rendered Filled Card. */
  skin: "pale" | "vivid";
  /** Sets horizontal layout. It changes the position of CardImage. */
  isHorizontal?: boolean;
  /** Reduces paddings and margins on Card layout. */
  isCompact?: boolean;
  /** Add a class name to make custom changes */
  className?: string;
}

export default function OutlineCard({
  children,
  color,
  skin,
  isHorizontal = false,
  isCompact = false,
  className = undefined,
}: OutlineCardProps) {
  return (
    <CardContext.Provider value={{ color, skin, cardType: "outline" }}>
      <article
        className={`lab-card lab-card--outline lab-card--${color} lab-card--${skin}
          ${isCompact ? " lab-card--compact" : ""}
          ${isHorizontal ? " lab-card--horizontal" : ""}
          ${className ? ` ${className}` : ""}
        `}
      >
        {children}
      </article>
    </CardContext.Provider>
  );
}
