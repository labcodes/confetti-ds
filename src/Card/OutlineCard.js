import React from "react";
import PropTypes from "prop-types";

import { CardContext } from "./contexts";
import { outlineCardChoices } from "./propTypes";

export default function OutlineCard({
  children,
  color,
  skin,
  isCompact,
  isHorizontal,
  className,
}) {
  return (
    <CardContext.Provider value={{ color, skin, cardType: "outline" }}>
      <article
        className={`lab-card lab-card--outline lab-card--${color} lab-card--${skin}
          ${isCompact ? " lab-card--compact" : ""}
          ${isHorizontal ? " lab-card--horizontal" : ""}
          ${className ? ` ${className}` : ""}
        `}
      >
        {children}
      </article>
    </CardContext.Provider>
  );
}

OutlineCard.propTypes = {
  /** Components that will be rendered in the Outline Card (CardImage, CardHeader, CardAction, etc.) */
  children: PropTypes.node.isRequired,
  /** Sets Outline Card's colors (mostly border colors) based on the chosen palette.  */
  color: PropTypes.oneOf(outlineCardChoices.color).isRequired,
  /** Sets the skin of the rendered Outline Card. */
  skin: PropTypes.oneOf(outlineCardChoices.skin).isRequired,
  /** Sets horizontal layout. It changes the position of CardImage. */
  isHorizontal: PropTypes.bool,
  /** Reduces paddings and margins on Card layout. */
  isCompact: PropTypes.bool,
  /** Add a class name to make custom changes */
  className: PropTypes.string,
};

OutlineCard.defaultProps = {
  isHorizontal: false,
  isCompact: false,
  className: undefined,
};
