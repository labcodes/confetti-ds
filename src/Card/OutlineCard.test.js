import React from "react";
import renderer from "react-test-renderer";

import OutlineCard from "./OutlineCard";

describe("OutlineCard", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineCard color="teal" skin="pale">
          Content
        </OutlineCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isCompact", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineCard color="teal" skin="pale" isCompact>
          Content
        </OutlineCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isHorizontal", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineCard color="teal" skin="pale" isHorizontal>
          Content
        </OutlineCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with extra className", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineCard color="teal" skin="pale" className="test-classname">
          Content
        </OutlineCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
