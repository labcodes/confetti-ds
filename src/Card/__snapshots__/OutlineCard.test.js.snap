// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineCard renders with base props 1`] = `
<article
  className="lab-card lab-card--outline lab-card--teal lab-card--pale
          
          
        "
>
  Content
</article>
`;

exports[`OutlineCard renders with isCompact 1`] = `
<article
  className="lab-card lab-card--outline lab-card--teal lab-card--pale
           lab-card--compact
          
        "
>
  Content
</article>
`;

exports[`OutlineCard renders with isHorizontal 1`] = `
<article
  className="lab-card lab-card--outline lab-card--teal lab-card--pale
          
           lab-card--horizontal
        "
>
  Content
</article>
`;
